### PR TITLE
feat(SCT-844): Add panel approval

### PR DESCRIPTION
**Pull request by anabebs**
**What**  
Screenshot from 2021-07-19 17-00-29
Screenshot from 2021-07-19 17-00-34
Screenshot from 2021-07-19 17-00-22

Adds frontend changes required for panel approval. This includes a new panel approval widget and dialogue box. 

**Why**  
Once a manager has approved a submission they will see a prompt to select whether the submission has also had panel approval or should be returned for edits.

**Anything else?**

The back end code is work in progress.

- Anything else in this PR that isn't covered by the what/why?
Tests will need to be written

### CODE REVIEW

File: components/ApprovalWidget/PanelApprovalWidget.tsx
Comment (jhackett1) [inline components/ApprovalWidget/PanelApprovalWidget.tsx:29]: try

"You cannot approve your own submissions. Ask a manager or colleague for help"

File: components/ApprovalWidget/PanelApprovalWidget.tsx
Comment (jhackett1) [inline components/ApprovalWidget/PanelApprovalWidget.tsx:41]: "Has the panel approved this work?"

File: components/ApprovalWidget/PanelApprovalWidget.tsx
Comment (jhackett1) [inline components/ApprovalWidget/PanelApprovalWidget.tsx:48]: "Yes, panel has approved"

**Comment by JohnFarrellDev** [comment]
It might be worth waiting for the API work to be on staging so we can fully check the functionality before merging.

We also need some tests.

File: components/ApprovalWidget/PanelApprovalWidget.tsx
Comment (jhackett1) [inline components/ApprovalWidget/PanelApprovalWidget.tsx:26]: "This submission needs panel approval"

File: components/ApprovalWidget/PanelApprovalWidget.tsx
Comment (jhackett1) [inline components/ApprovalWidget/PanelApprovalWidget.tsx:38]: "This submission needs panel approval"

File: components/ApprovalWidget/PanelApproveDialogue.tsx
Comment (jhackett1) [inline components/ApprovalWidget/PanelApproveDialogue.tsx:36]: "Are you sure you want to mark this submission as panel approved?"

File: components/ApprovalWidget/PanelApproveDialogue.tsx
Comment (jhackett1) [inline components/ApprovalWidget/PanelApproveDialogue.tsx:53]: replace with mandatory checkbox with "Yes, the panel met and approved this work"

File: components/ApprovalWidget/PanelApproveDialogue.tsx
Comment (jhackett1) [inline components/ApprovalWidget/PanelApproveDialogue.tsx:57]: add `useState` linked to the checkbox to disable this conditionally

File: components/ApprovalWidget/PanelApproveDialogue.tsx
Comment (jhackett1) [inline components/ApprovalWidget/PanelApproveDialogue.tsx:59]: Yes, mark as approved

**Comment by jhackett1** [comment]
should include "marked as panel approved by x" on the history timeline

File: lib/submissions.ts
Comment (JohnFarrellDev) [inline lib/submissions.ts:190]: Could the string value be `panel-approved` instead. It would be nicer for the backend.

Comment (anabebs) [reply lib/submissions.ts:190]: sure

**Comment by jhackett1** [comment]
Capture

**Comment by JohnFarrellDev** [comment]
I noticed `RejectDialog.tsx` exported component is called `ApproveDialog`, it's just a copy and paste error from before but should be named correctly

File: components/ApprovalWidget/PanelApproveDialog.tsx
Comment (JohnFarrellDev) [inline components/ApprovalWidget/PanelApproveDialog.tsx:87]: This should be disabled if the checkbox hasn't been ticked for "The panel met and approved this work" right?

Comment (JohnFarrellDev) [reply components/ApprovalWidget/PanelApproveDialog.tsx:87]: The button doesn't submit but visually would be nice to show it is disabled

**Review by JohnFarrellDev — APPROVED** [review]

File: components/ApprovalWidget/PanelApproveDialog.tsx
Comment (jhackett1) [reply components/ApprovalWidget/PanelApproveDialog.tsx:87]: that's taken care of by the validation schema on the form that surrounds this

Comment (JohnFarrellDev) [reply components/ApprovalWidget/PanelApproveDialog.tsx:87]: I noticed when I went to the dialog the button wasn't disabled even though the checkbox had not been ticked yet. Just a visual bug.

**Pull request merged**
